### ISSUE TITLE
Seeding and randomness in Megatron Energon

### ISSUE THREAD

**Issue by bernardhan33** [NVIDIA/Megatron-Energon#76]
Keeping the configuration of the Megatron Energon data loaders consistent between runs makes data loaders loading the data in the same sequences. To elaborate, using the following code

```python3
train_dataset = get_train_dataset(
      cfg.model.energon.path,
      batch_size=cfg.model.micro_batch_size,
      task_encoder=task_encoder,
      worker_config=worker_config,
      max_samples_per_sequence=None,
      packing_buffer_size=None,
      shuffle_buffer_size=100,
      split_part='train',
  )
  data_loader = get_savable_loader(train_dataset, worker_config=worker_config)
```

and run it repeatedly on the same rank makes the data loader load the same sequence of data (although the data itself has been shuffled with the expectation of using 100 samples in the buffer, as defined by `shuffle_buffer_size`)

Is this expected? Is there a way to set up a different seed as input to Megatron Energon?

**Comment by philipp-fischer** [comment]
Hi @bernardhan33.  Indeed this reproducibility is one of the core features of energon :) 
If you want to change the order, you need to set a seed offset in your worker config: https://github.com/NVIDIA/Megatron-Energon/blob/develop/src/megatron/energon/worker.py#L52

**Comment by bernardhan33** [comment]
Thank you for your prompt response! Closing.
